cytoscape-edgehandles.js

```diff
@@ -619,12 +765,21 @@ SOFTWARE.
               return; // nothing to do :(
             }
 
+            const hit = hitTest(targets, { x: mx, y: my });
+
             // just remove preview class if we already have the edges
             if( !src && !tgt ) {
               if( !preview && options().preview ) {
                 added = cy.elements( '.edgehandles-preview' ).removeClass( 'edgehandles-preview' );
-                options().complete( source, targets, added );
-                source.trigger( 'cyedgehandles.complete' );
+
+                if ( !hit ) {
+                  added.remove();
```

Comment:
If it changes the previously expected behaviour, it might be good to have this feature enabled by an option.  That way it could be disabled by default so that it doesn't break existing code.